### ISSUE TITLE
`HyperParameterOptimizer: TypeError: 'NoneType' object is not iterable`

### ISSUE THREAD

**Issue by iirekm** [allegroai/clearml#243]
When I run this simple code:

```python
task = Task.init(project_name='...', task_name='opt', task_type=Task.TaskTypes.optimizer)
print(task.get_parameters())
if task.get_parameter("General/opt/optimizer_class", None) is None:
    task.execute_remotely(queue_name="default")
    opt = HyperParameterOptimizer(
        optimizer_class=OptimizerOptuna,
        base_task_id=task.id,
        hyper_parameters=[
            UniformIntegerParameterRange('General/x1', min_value=-100, max_value=100),
            UniformIntegerParameterRange('General/x2', min_value=-100, max_value=100),
        ],
        objective_metric_title="performance",
        objective_metric_series="loss",
        objective_metric_sign="min_global",
        total_max_jobs=3,
        max_iteration_per_job=100,
    )
    opt.start()
    opt.wait()
    opt.stop()
else:
...
```

I get:

```
Traceback (most recent call last):
File "hyperopt-demo.py", line 27, in
max_iteration_per_job=100,
File "/home/.../.trains/venvs-builds/3.7/lib/python3.7/site-packages/trains/automation/optimization.py", line 937, in __init__
optimizer_class=optimizer_class, hyper_param_configuration=hyper_parameters, **opts)
File "/home/.../.trains/venvs-builds/3.7/lib/python3.7/site-packages/trains/automation/optimization.py", line 1227, in _connect_args
self._task.connect_configuration(configuration_dict)
File "/home/.../.trains/venvs-builds/3.7/lib/python3.7/site-packages/trains/task.py", line 1052, in connect_configuration
else self._get_model_config_dict())
TypeError: 'NoneType' object is not iterable
```

However when I run it locally (without `task.execute_remotely(queue_name="default")`) it works perfectly, but local hyperopt makes no sense, because it takes a lot of time. Because it crashes on agent, I can't even debug it easily.

**Comment by bmartinn** [comment]
Hi @iirekm 
I'm not sure I understand the rational behind the code...
If you need to test whether you are running remotely (i.e. with `trains-agent`) :
```python
from trains import Task

if not Task.running_locally():
    pass  # do something here 
```

Also, I'd like to remove some of the mystery behind "execute_remotely", basically it will ask, if we are not executed by `trains-agent` enqueue the current running Task (i.e. ourselves) and leave the process, if we are being executed by the `trains-agent` do nothing.
This means there is no need to add an extra `If` before the execute_remotely call, it will do nothing when called under `trains-agent`. Also notice that when executed locally, it will literally call `os.exit()` and leave the process, so the line after it will never be executed locally.

**Comment by iirekm** [comment]
I know if I'm running locally or not. The problem is that under agent the above code crashes, without agent (eg with IntelliJ) it works.

Workaround for this: add To `HyperParameterOptimizer` param `auto_connect_task=False`.

**Comment by bmartinn** [comment]
Right, I see why, `HyperParameterOptimizer` stores it state on the Task itself (so that later you could clone and edit the `HyperParameterOptimizer` and for example add/modify another parameter to optimize, from the UI.
In the code example, the `HyperParameterOptimizer` was never called on the original Task (since it was called after `executed_remotely`, so when executed by the trains-agent it crashed (which obviously should be fixed to: "output a warning").
To fix it, you can just move the `executed_remotely` line to after  the `HyperParameterOptimizer` initialization

**Comment by jkhenning** [comment]
Just updating this was released in ClearML/Trains v0.16.4 🙂
